src/Microsoft.TestPlatform.CrossPlatEngine/Client/ProxyExecutionManagerWithDataCollection.cs

```diff
@@ -134,6 +134,31 @@ public override void Cancel()
             }
         }
 
+        public override int LaunchProcessWithDebuggerAttached(TestProcessStartInfo testProcessStartInfo)
+        {
+            if (this.dataCollectionEnvironmentVariables != null)
+            {
+                if (testProcessStartInfo.EnvironmentVariables == null)
+                {
+                    testProcessStartInfo.EnvironmentVariables = new Dictionary<string, string>();
+                }
+
+                foreach(var envVariable in this.dataCollectionEnvironmentVariables)
+                {
+                    if (testProcessStartInfo.EnvironmentVariables.ContainsKey(envVariable.Key))
+                    {
+                        testProcessStartInfo.EnvironmentVariables[envVariable.Key] = envVariable.Value;
+                    }
+                    else
+                    {
+                        testProcessStartInfo.EnvironmentVariables.Add(envVariable.Key, envVariable.Value);
```

Comment:
We can just keep line 150, it will not throw if key doesn't exist and will replace the value if it exists.